src/geometry.cpp

```diff
@@ -294,27 +296,32 @@ find_cell(Particle* p, bool use_neighbor_lists)
 void
 cross_lattice(Particle* p, const BoundaryInfo& boundary)
 {
-  auto& lat {*model::lattices[p->coord_[p->n_coord_-1].lattice]};
+  auto& coord {p->coord_[p->n_coord_ - 1]};
+  auto& lat {*model::lattices[coord.lattice]};
 
   if (settings::verbosity >= 10 || simulation::trace) {
     std::stringstream msg;
     msg << "    Crossing lattice " << lat.id_ << ". Current position ("
-         << p->coord_[p->n_coord_-1].lattice_x << ","
-         << p->coord_[p->n_coord_-1].lattice_y << ","
-         << p->coord_[p->n_coord_-1].lattice_z << ")";
+        << coord.lattice_x << "," << coord.lattice_y << ","
+        << coord.lattice_z << "). r=" << p->r();
     write_message(msg, 1);
   }
 
   // Set the lattice indices.
-  p->coord_[p->n_coord_-1].lattice_x += boundary.lattice_translation[0];
-  p->coord_[p->n_coord_-1].lattice_y += boundary.lattice_translation[1];
-  p->coord_[p->n_coord_-1].lattice_z += boundary.lattice_translation[2];
-  std::array<int, 3> i_xyz {p->coord_[p->n_coord_-1].lattice_x,
-                            p->coord_[p->n_coord_-1].lattice_y,
-                            p->coord_[p->n_coord_-1].lattice_z};
+  coord.lattice_x += boundary.lattice_translation[0];
+  coord.lattice_y += boundary.lattice_translation[1];
+  coord.lattice_z += boundary.lattice_translation[2];
+  std::array<int, 3> i_xyz {coord.lattice_x, coord.lattice_y, coord.lattice_z};
 
   // Set the new coordinate position.
-  p->r_local() = lat.get_local_position(p->coord_[p->n_coord_-2].r, i_xyz);
+  const auto& upper_coord {p->coord_[p->n_coord_ - 2]};
+  const auto& cell {model::cells[upper_coord.cell]};
+  Position r = upper_coord.r;
+  r -= cell->translation_;
+  if (!cell->rotation_.empty()) {
```

Comment:
It _could_. This would make the code a bit cleaner here by hiding the conditional check. However, it could add some overhead to the non-rotated case, where it would basically set `r = r`. Not sure if the compiler is smart enough to see that it is a no-op.

Reply:
Fair enough. Yeah I'd be surprised if that were the case. We can always update it later if needed.

Reply:
Yeah I played around with this in your branch and in an isolated example. It didn't seem to make much of a difference. I'm good w it either way.